configure.py

```diff
@@ -1053,6 +1053,7 @@ def host_arch_cc():
   k = cc_macros(os.environ.get('CC_host'))
 
   matchup = {
+    '_LP64'       : 'arm64',
```

Comment:
`_LP64` is not arm64 specific so this is going to break non-arm 64-bit platforms.

Reply:
True, I just came to the same understanding while you were commenting. I wrote a sum-up of the issue in my updated PR desc.

Reply:
The true cause of the issue is that when we assume that the parsed output of e.g. `${CC_host} -dM -E -` will contain the host arch at a specific index like we do here:

https://github.com/nodejs/node/blob/d23af0d5aef937cf1b1180b285cbc4eebbf1e7ee/configure.py#L1072

Just isn't there because the output is not standardized (as it seems) or other circumstances make the platform write stuff like `_LP64` there... well, then we're going to assume the wrong host_arch here.

Any ideas? :)

Reply:
I could just put a special override if-condition to check for M1 as a platform instead of relying on the `cc` output but it is such a nice abstract impl, that it seems a bit hacky to do that. However, Node.js not being able to compile on M1 is quite a blocker... so maybe...

Reply:
Unfortunately, in the case of building with a fat binary, even running `uname -p` or `uname -m` gives me `i386` and `x86_64`. This is probably because Rosetta2 emulates "I'm running on an x86_64 machine". There is only one way to escape this rabbit hole (as far as I can see atm): Running `uname -v` which gives you the kernel version string as `Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:20 PDT 2021; root:xnu-7195.141.6~3/RELEASE_ARM64_T8101`. 

This can *not* reproduced by running `uname -p` or `uname -m` on the command line of an M1 machine. It can only reproduced when the parent process that executes these shell commands is emulated via Rosetta.

As we can see, the `uname -v` kernel version string includes the ARM64 release tag which indicates that the kernel is actually running on an ARM platform and the toolchain is running as a universal binary, translated via Rosetta2.